### DISCUSSION TITLE
Dynamic routes and rewrites not working as expected

### DISCUSSION THREAD

**Discussion by pueyo5** [vercel/next.js#16040] [Help]
I'm using rewrites with dynamic routes, and it fires a "Unhandled Runtime Error".

My folder structure is:
```
| - index.js
| - /cars
    | - /[brand]
```

And I want to access the brand page using /cars-audi instead of /cars/audi, that's why I'm using rewrites (both next and vercel).
```
async rewrites() {
    return [
        { source: "/cars-:brand", destination: "/cars/:brand" }
    ]
}
```

Then, on the brand page, I use getStaticPaths with fallback true and getStaticProps with revalidate.

```
export async function getStaticPaths() {
  return { paths: [], fallback: true }
}

export async function getStaticProps({ params }) {
  const res = await getBrandData(params.brand)
  const item = await res.json()

  return { props: { item, notFound: res.status === 404 }, revalidate: 60 }
}
```

The problem is that the expected value of "params" is `{ brand: 'audi' }`, but it's current value is `{ brand: 'audi?brand=audi' }
`

And it is trying to fetch the wrong json file:
`http://localhost:3000/_next/data/development/cars/audi%3Fbrand=audi.json`

Am I doing something wrong or it's a bug?

Thanks

**Comment by mAAdhaTTah** [comment]
There was a big related to this: https://github.com/vercel/next.js/issues/15626 Not 100% sure it's released but it was merged.
